Add tests for verifying action key on hyrolo matches

matsl opened the pull request:
# What

Add tests for verifying action key on hyrolo matches

# Why

I noticed a problem with kotl files if the first line contains a slash
'/'.  So added a few tests to explore that problem. The problematic
test is marked as expected failure.

# Note

The problem is that the kotl-files contents "display" is changed and the file is not a any longer a valid kotl-file. The file stored on disk is not changed and is still a valid kotl-file. So something happend when the action key is used to move there when there is a a slash in the line!?

rswgnu commented on test/hyrolo-tests.el:
```diff
@@ -1513,6 +1513,108 @@ body
       (hy-delete-files-and-buffers hyrolo-file-list))))
 
+(ert-deftest hyrolo-tests--goto-org-match ()
+  "Verify that moving to a match reaches the target buffer."
```
Need to change the doc string to match the test for each of the fun tions you added to this buffer.

matsl replied:
I will fix. I wanted to you too see it so I hacked and pushed away! ;-)

rswgnu commented on test/hyrolo-tests.el:
```diff
@@ -1513,6 +1513,108 @@ body
+         (hyrolo-file-list (list kotl-file1)))
+    (unwind-protect
+        (progn
+          (hyrolo-grep "h/kotl")
```
/  is the child separator in hyrolo searches, so this means find the ‘h’ node and then a child node starting with ‘kotl’.  Should be documented in the manual.  So the test needs rewriting.  Maybe we should not have the same behavior in hyrolo-fgrep and hyrolo-grep.

matsl replied:
Ah I see. OK. To bad the point I was trying to make got lost because it got masked by that feature. So you are right that the test case needs to be rewritten but not because of the child separator. It is a problem with having a / in the first line in the cell if there is a match on that line. To late this night to rewrite it so I will redesign it tomorrow to better prove the point. Or I can show you a real life example on Sunday.

matsl commented:
@rswgnu Updated docstrings and reworked the tests slightly. Most important. Removed the slash in the search since that has special meaning i hyrolo but kept it in the contents of the cells. There are test cases both for slash in the first line in the cell and in the second line in the cell. There is different behavior.
